src/bptl/credentials/forms.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Dict, List
+
+from django import forms
+from django.db.models import BLANK_CHOICE_DASH
+from django.utils.translation import gettext_lazy as _
+
+from zgw_consumers.concurrent import parallel
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+
+from bptl.work_units.zgw.utils import get_paginated_results
+
+from .models import App
+
+
+def _get_applications(service: Service) -> List[Dict[str, Any]]:
+    client = service.build_client()
+    results = get_paginated_results(client, "applicatie")
+    return results
+
+
+class AppForm(forms.ModelForm):
+    autorisaties_application = forms.ChoiceField(
+        label=_("Autorisaties API app"),
+        required=False,
+        help_text=_(
+            "Application in Autorisaties API. If selected, label and App ID "
+            "can be left blank."
+        ),
+    )
+
+    class Meta:
+        model = App
+        fields = (
+            "autorisaties_application",
+            "label",
+            "app_id",
+        )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._set_auth_api_choices()
+
+    def _set_auth_api_choices(self):
+        autorisatie_apis = Service.objects.filter(api_type=APITypes.ac)
+
+        with parallel() as executor:
+            applications = executor.map(_get_applications, autorisatie_apis)
+
+        all_apps = sum(applications, [])
+        all_apps = sorted(all_apps, key=lambda app: app["label"])
+
+        # if there's no data, no point in showing the field at all
+        if not all_apps:
+            del self.fields["autorisaties_application"]
+
+        self.fields["autorisaties_application"].choices = BLANK_CHOICE_DASH + [
```

Comment:
lines 57-64 should be indented

Reply:
Mmm, it doesn't look like it's as easy as just indenting.

Reply:
right, good catch. Need to take a good look at this!

Reply:
And is it correct that you use Autorisatie API only to get the list of application names? But you don't use credentials since it would not be secure? But in this case why use Autorisatie API at all?

Reply:
Fixed via f0520669d575de183fa6f62e009e24f6d6d64fbe

Reply:
> And is it correct that you use Autorisatie API only to get the list of application names? But you don't use credentials since it would not be secure? But in this case why use Autorisatie API at all?

Correct. The `app_id` can be any string, but it makes everyone's life easier if we re-use the Autorisaties API URL as ID - as the application can also retrieve that itself from the Autorisaties API so they don't need to exchange that before-hand/make it configurable.

Autorisaties API does not expose credentials of apps, that's a security can of worms I don't want to open.

We use it as convenience - if you can select the app from a pre-defined list, this is less admin overhead.